File: BackspaceStringCompare.java
//https://leetcode.com/problems/backspace-string-compare/

class Solution {
    public boolean backspaceCompare(String s, String t) {
        return compute(s).equals(compute(t));
    }

    public String compute(String s) {
        StringBuilder ans = new StringBuilder();
        int skip = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) == '#') {
                skip++;
            } else if (skip > 0) {
                skip--;
            } else {
                ans.append(Character.toString(s.charAt(i)));
            }
        }
        return ans.toString();
    }
}

File: Easy/BackspaceStringCompare/BackspaceStringCompare.cpp
// https://leetcode.com/problems/backspace-string-compare/

class Solution {
public:
    bool backspaceCompare(string s, string t) {
        return format(s)==format(t);
    }

    string format(string str){
        string result = "";
        for(int i = 0; i<str.length(); i++){
            if(str[i]!='#'){
                result.push_back(str[i]);
            }else if(result.length()>0){
                result.pop_back();
            }
        }

        return result;
    }
};